src/formats.ts

```diff
@@ -169,6 +191,19 @@ function uri(str: string): boolean {
   return NOT_URI_FRAGMENT.test(str) && URI.test(str)
 }
 
+function validateInteger(bits: number): (value: number | string) => boolean {
+  const max = BigInt(2) ** BigInt(bits - 1)
+  const min = max * BigInt(-1)
+  return (value) => Number.isInteger(+value) && BigInt(value) <= max && BigInt(value) >= min
```

Comment:
... and then no need for `+`

Also, the way max is defined it should be "<" max (strictly less)?

Reply:
Yup: 
min = -2^31
max = (2^31)-1 
so it needs to be strictly less